### DISCUSSION TITLE
Requests on MongoDB & Learning material

### DISCUSSION THREAD

**Discussion by AugCei** [BHoM/BHoM#1384] [Q&A] [answered]
Hi, 

I recently started to have a look at BHoM to asses whether it would help fulfilling some of our needs in Civil structures. (And so far so good, very exciting project that we would like to contribute to in the future). 

I'm facing an issue related to requests on the Mongo Adapter: Seems that the only "Adapter" type of request is about the collections names and the "Data" type of request doesn't work when pulling from MongoDB. Any Idea how I can pull only certain data from MongoDB and not the full set? Im currently pulling the whole thing and then filtering data but have the feeling that this is not a very good practice. 

Furthermore, Any idea where I could find some good learning material to understand a bit better BHoM and the various libraries that it has? I Would like to know if there would be another way before diving into the depth of GitHub that seems for a newbie like me a very big step outside of the comfort zone. If no, ill just dive into that and brace for impact! 

Thank you and hopefully this was the right place to ask my question! 

Augustin
![BHoM-Question](https://user-images.githubusercontent.com/104192769/165522224-6e3916a2-a7a9-4403-8420-195095724431.png)

**Comment by FraserGreenroyd** [comment]
Hi @AugCei , welcome to the BHoM!

In general, the best place to start start would be the wiki available [here](https://github.com/BHoM/documentation/wiki) - this covers the highlevel philosophy of the BHoM and a few ins/outs of how things work (including [the interaction between oM/Engine/Adapter](https://github.com/BHoM/documentation/wiki/Structure-of-the-BHoM) ).

For Mongo specifically, there is some documentation available within the [Mongo Toolkit wiki](https://github.com/BHoM/Mongo_Toolkit/wiki) that may help? Alternatively, I think @adecler (who is currently away for a couple of weeks), @IsakNaslundBh and @alelom have experience in using Mongo Toolkit and may be able to provide assistance too.

For Civils bits, we do have a skeleton of a [Civil3D Toolkit](https://github.com/BHoM/Civil3D_Toolkit) - but this is very basic and needs some love and attention - @cesarecaoduro has been looking into some Civil3D bits from memory.

As for diving in, I personally think it's a good way to go, there's a lot of stuff available here generated over the years and plenty of gems to find, but if you do get stuck at any point feel free to reach out on a discussion or issue and someone will hopefully be along to help 😄

**Comment by AugCei** [comment, marked as answer]
Hi @FraserGreenroyd, Thanks for your quick answer! 

The [Mongo Toolkit wiki ](https://github.com/BHoM/Mongo_Toolkit/wiki) had indeed all the information, more specifically [MongoDB Queries in a Nutshell](https://github.com/BHoM/Mongo_Toolkit/wiki/MongoDB-Queries-in-a-Nutshell)! 

Id be interested in discussion futher about Civil structures if somebody is having some interest about that! We are starting from scratches but have quiet a lot of interest in getting this growing. 

Ill get to dive into the Wiki then! Hopefully this will get clearer as I get lost in there!
